sample/transforms.go

```diff
@@ -95,51 +104,45 @@ func topK(ts []token, k int) []token {
 		result[i] = heap.Pop(&h).(token)
 	}
 
-	return result
+	*ts = result
 }
 
 // topP limits tokens to those with cumulative probability p
-func topP(ts []token, p float32) []token {
+func topP(ts *[]token, p float32) {
```

Comment:
`topP` and friends below can stay a regular slice parameter

```suggestion
func topP(ts []token, p float32) {
```

Reply:
Cool - so topK will be the only transform which modifies the array. The rest will just be returning views into the underlying array